src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -173,37 +171,41 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     void TermControl::UpdateSettings(Settings::IControlSettings newSettings)
     {
         _settings = newSettings;
+        auto weakThis{ get_weak() };
 
         // Dispatch a call to the UI thread to apply the new settings to the
         // terminal.
-        _root.Dispatcher().RunAsync(CoreDispatcherPriority::Normal, [this]() {
-            // Update our control settings
-            _ApplyUISettings();
+        _root.Dispatcher().RunAsync(CoreDispatcherPriority::Normal, [weakThis]() {
+            if (auto control{ weakThis.get() })
+            {
+                // Update our control settings
+                control->_ApplyUISettings();

```

Comment:
not 100% sure on how i feel about indirecting through `control->` every time; in debug builds each one will be a call to `winrt::com_ptr::operator->()`...
but i do not know how much, if at all, I care. Thoughts?

Reply:
(we could capture `weakThis,this` and then only _use_ `this` implicitly if the weak got locked. @zadjii-msft?)

Reply:
I prefer the way the `weakThis,this` lambdas read. All the `control->` just clutters things, not mention your comment about `winrt::com_ptr::operator->()`. WinRT could almost use a class to support this form of lambda. It would be extensible to all the places where we're having trouble with the `{ this, pointer-to-member }` event format. Anyways, I'll include the fix to these in the next revision commit.

Reply:
I think I prefer it like this - not capturing `this` at all makes it harder for us to accidentally do the wrong thing.

Reply:
I guess there's a "convenient" middle ground...

```
if (auto strongThis{ weakThis.get() })
{
    auto& that{ *strongThis };
    that->x;
    that->y;
}
```

I _am_ worried about the constant indirection through `operator->`... but at the same time, `WRL::ComPtr` _literally is that for all com calls_, sooooooooo maybe i should not be worried?

Reply:
It's just that this is a member function, but we have to write it in a weird way that makes it not look like a member function.

Reply:
Let me know which form to go with, and I'll include it in the next commit.